src/api/collection.ts

```diff
@@ -15,6 +22,43 @@ export class Collection {
     this.config = config;
   }
 
+  /**
+   * Creates a new collection within the specified account
+   *
+   * @param args.creator the account of the collection's creator
+   * @param args.description the description of the collection
+   * @param args.name the name of the collection
+   * @param args.uri the URI to additional info about the collection
+   *
+   * The parameters below are optional.
+   * @param args.maxSupply controls the max supply of the tokens - defaults MAX_U64_BIG_INT
+   * @param args.mutableDescription controls mutability of the collection's description - defaults true
+   * @param args.mutableRoyalty controls mutability of the collection's description - defaults true
+   * @param args.mutableUri controls mutability of the collection's URI - defaults true
+   * @param args.mutableTokenDescription controls mutability of the token's description - defaults true
+   * @param args.mutableTokenName controls mutability of the token's name - defaults true
+   * @param args.mutableTokenProperties controls mutability of token's properties - defaults true
+   * @param args.mutableTokenUri controls mutability of the token's URI - defaults true
+   * @param args.tokensBurnableByCreator controls whether tokens can be burnable by the creator - defaults true
+   * @param args.tokensFreezableByCreator controls whether tokens can be frozen by the creator - defaults true
+   * @param args.royaltyNumerator the numerator of the royalty to be paid to the creator when a token is transferred - defaults 0
+   * @param args.royaltyDenominator the denominator of the royalty to be paid to the creator when a token is transferred -
+   *    defaults 1
+   *
+   * @returns A SingleSignerTransaction that when submitted will create the collection.
+   */
+  async createCollectionTransaction(
+    args: {
+      creator: Account;
+      description: string;
+      name: string;
+      uri: string;
+      options?: GenerateTransactionOptions;
+    } & CreateCollectionOptions,
+  ): Promise<SingleSignerTransaction> {
+    return createCollectionTransaction({ aptosConfig: this.config, ...args });
+  }
+
```

Comment:
Returning a SingleSigner may be weird if we also support Fee payer, but I'm not sure how that model is going to look cc @0xmaayan